### PR TITLE
Buy Me a Coffee

### PR DESCRIPTION
**Pull request by gkapkowski**
Added "Buy Me a Coffee" button + various fixes and improvements

Screenshot from 2023-05-06 20-40-49

### CODE REVIEW

File: login/login/frontend/src/App.tsx [hunk 1]
```diff
@@ -207,34 +206,46 @@ export const App: React.FC<AppProps> = ({
   let providers = [injectedProvider, ...walletProviders];
   let origin = loginProvider.loginData.clientId;
   let restrictions = loginProvider.loginData.restrictions;
+  const LoginPage = lazy(
+    () => import(/* webpackPreload: true */ "./components/LoginPage")
```
Comment (nmkedziora) [inline login/login/frontend/src/App.tsx:210]: You could also use magic comment for naming a chunk. Try:
```
/*webpackChunkName: "login-page" */ 
```

File: login/login/frontend/src/App.tsx [hunk 2]
```diff
@@ -207,34 +206,46 @@ export const App: React.FC<AppProps> = ({
...
+  );
+  const ConnectWalletPage = lazy(
+    () => import(/* webpackPreload: true */ "./components/ConnectWalletPage")
```
Comment (nmkedziora) [inline login/login/frontend/src/App.tsx:213]: ditto

File: login/login/frontend/src/backgrounds/Default.tsx [hunk 3]
```diff
@@ -10,21 +10,11 @@ import Paper from "@material-ui/core/Paper";
...
-    <Typography variant="body2" color="textSecondary" align="center">
-      {"More information at "}
-      <Link color="inherit" href="https://cryptoauth.io/">
-        cryptoauth.io
-      </Link>{" "}
-      {"."}
-    </Typography>
-  );
-}
+const App = lazy(() => import(/* webpackPreload: true */ "../App"));
```
Comment (nmkedziora) [inline login/login/frontend/src/backgrounds/Default.tsx:17]: ditto

File: login/login/frontend/src/backgrounds/FwB.tsx [hunk 4]
```diff
@@ -1,8 +1,10 @@
-import React, { useState } from "react";
+import React, { useState, lazy, Suspense } from "react";
 import { makeStyles, Theme } from "@material-ui/core/styles";
-import { App } from "../App";
 import { ILoginProvider } from "../services/loginProvider";
 import { ProviderList } from "../components/ConnectWalletPage";
+import { Loader } from "../components/Loader";
+
+const App = lazy(() => import(/* webpackPreload: true */ "../App"));
```
Comment (nmkedziora) [inline login/login/frontend/src/backgrounds/FwB.tsx:7]: ditto

File: login/login/frontend/src/components/ConnectWalletPage.tsx [hunk 5]
```diff
@@ -195,7 +195,7 @@ export const ConnectWalletPage: React.FC<ConnectWalletPageProps> = ({
               container
               xs={12}
               className={classes.help}
-              //justify="center"
+              //justifyContent="center"
```
Comment (nmkedziora) [inline login/login/frontend/src/components/ConnectWalletPage.tsx:198]: Should this comment be here?

File: login/login/frontend/src/components/LoginPage.tsx [hunk 6]
```diff
@@ -173,7 +173,7 @@ export const LoginPage: React.FC<LoginPageProps> = ({
               container
               xs={12}
               className={classes.help}
-              //justify="center"
+              //justifyContent="center"
```
Comment (nmkedziora) [inline login/login/frontend/src/components/LoginPage.tsx:176]: Should this comment be here? Can you remove or uncomment it?

File: login/login/frontend/src/embed/index.ts [hunk 7]
```diff
@@ -10,9 +10,9 @@ import { Window, getIdentities } from "./utils";
   async function main() {
     try {
       const loginProvider: IProvider = {
-        init: (await import(/* webpackPrefetch: true */"./init")).init,
-        login: (await import(/* webpackPrefetch: true */"./login")).login,
-        connect: (await import(/* webpackPrefetch: true */"./connect")).connect,
+        init: (await import(/* webpackPreload: true */"./init")).init,
+        login: (await import(/* webpackPreload: true */"./login")).login,
+        connect: (await import(/* webpackPreload: true */"./connect")).connect,
```
Comment (nmkedziora) [inline login/login/frontend/src/embed/index.ts:15]: Are you sure you want to change this? 👇 


> The `prefetch` browser hint can be used to fetch resources that may be needed some time soon, but not immediately on the initial load.

> The `preload` browser hint can be used to fetch resources that are critical to the current navigation, such as fonts or images are instantly (not longer than 3 seconds after the initial load) visible on a landing page.

File: login/login/frontend/src/index.tsx [hunk 8]
```diff
@@ -1,9 +1,8 @@
-import React from "react";
-import ReactDOM from "react-dom";
+import React, { StrictMode } from "react";
+import { createRoot } from "react-dom/client";
 import CssBaseline from "@material-ui/core/CssBaseline";
 import { ThemeProvider } from "@material-ui/core/styles";
-import { App } from "./App";
-import reportWebVitals from "./reportWebVitals";
+//import reportWebVitals from "./reportWebVitals";
```
Comment (nmkedziora) [inline login/login/frontend/src/index.tsx:5]: Should it stay here?

**Comment by gkapkowski** [comment]
Todo:
- change login page name from client id to some metadata, it's random uid now
- 
Screenshot from 2023-07-29 19-20-38

**Comment by gkapkowski** [comment]
TODO:
- Page title is still in uuid format

Screenshot from 2023-11-05 21-13-52

**Comment by gkapkowski** [comment]
This is blocked by hydra change to use uuids as client ids: https://github.com/ory/hydra/issues/2911
Need to wait for new version which reverts this decision.

**Comment by gkapkowski** [comment]
OR not, looking at MR code all of the existing client should work as is

**Pull request merged**